api_server.py

```
@@ -160,9 +162,11 @@ def check_rules(available_ids: List[int]) -> Dict:
         }
     with get_db() as conn:
         placeholders = ",".join("?" * len(available_ids))
-        rows = conn.execute(
-            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})", available_ids
-        ).fetchall()
+        # Use nosec B608 as placeholders are generated from safe integer IDs
+        query = (
+            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})"  # nosec B608
```

Ruff (0.15.5)

[error] Possible SQL injection vector through string-based query construction

(S608)